base/iddict.jl

```diff
@@ -162,6 +162,7 @@ function get!(default::Callable, d::IdDict{K,V}, @nospecialize(key)) where {K, V
         if !isa(val, V)
             val = convert(V, val)
         end
+        val::V
```

Comment:
Does it still work if you put this on the `convert(V, val)` call instead? That seems better.